pkg/hub/submarineragent/controller.go

```diff
@@ -375,24 +379,33 @@ func (c *submarinerAgentController) syncSubmarinerConfig(ctx context.Context,
 
 	// prepare submariner cluster environment
 	errs := []error{}
-	cloudProvider, preparedErr := cloud.GetCloudProvider(c.restConfig, c.kubeClient, c.manifestWorkClient, c.dynamicClient,
-		c.eventRecorder, managedClusterInfo, config)
-	if preparedErr == nil {
-		preparedErr = cloudProvider.PrepareSubmarinerClusterEnv()
-	}
+	var condition  metav1.Condition
+	if config.Status.ManagedClusterInfo.Platform == "AWS" {
```

Comment:
This is checking only for AWS. What about GCP/VmWare? Why are we handling AWS and GCP (or VmWare) separately? I feel we they should be handled at the same place.